WcaOnRails/app/javascript/edit-schedule/EditSchedule.jsx

```diff
@@ -51,6 +52,11 @@ export default class EditSchedule extends React.Component {
   componentDidMount() {
     wca.datetimepicker();
     window.addEventListener("beforeunload", this.onUnload);
+    $(venuesEditPanelSelector).find('.panel-collapse').on('shown.bs.collapse', () => {
```

Comment:
I have no idea why/how google maps does this, but basically leaflet need a height to be able to render all the tiles properly, and it doesn't exist when the element has a `display:none` property.

It's basically a behavior I observed, and googling around shown it was a "common" issue where the solution was to call `invalidateSize` on the map.

Reply:
I finally had some time to dig into this. It looks like the Google Maps javascript library actually is able to register an event listener to detect when the map becomes visible/invisible. The trick seems to be to create an empty `<iframe>`, and adds a resize listener to the iframe. See below:

![image](https://user-images.githubusercontent.com/277474/48921875-b9b4cf00-ee57-11e8-8414-e74b26ccd48a.png)

This `OG` function gets called when the `iframe.contentWindow`'s `resize` event fires, and then it goes and updates the size of the google map accordingly.

I am able to use the same trick over in the leaflet world, and detect that the map is appearing/disappearing!

Unfortunately, I can't think of a good way to actually integrate this workaround into the code as written, because sometimes we use leaflet directly, and sometimes we use react-leaflet. I think that would require writing two different wrappers that implement this trick... what do you think is best?

Reply:
Would you mind giving me some more details on how you set it up?
I did a couple of tries by adding this invisible iframe to the leaflet's div, but I can't seem to get any "resize" event fired when expanding/collapsing the venues panel :s

(I do get the "load" event; I get the "resize" event when I change the viewport (by slightly resizing the developer tools frame for instance), but I don't get the "resize" when only manipulating the panels)

I'll push my rebase now.

Reply:
Hmm... does this work? https://www.jflei.com/misc/pr-3025-iframeing/index.html

![2019-02-24_12-19-35_dalinar](https://user-images.githubusercontent.com/277474/53304811-7ab76080-382e-11e9-97c2-593d028f3cb3.gif)